lib/sprockets/less/v3/importer.rb

```diff
@@ -0,0 +1,224 @@
+        def resolve_glob(context, glob, base_path)
+          base_path      = Pathname.new(base_path)
+          path_with_glob = base_path.dirname.join(glob).to_s
+
+          Pathname.glob(path_with_glob).sort.select do |path|
+            asset_requirable = asset_requirable?(context, path)
+            path != context.pathname && asset_requirable
+          end
+        end
+
+        # Returns all of the possible paths (including partial variations)
+        # to attempt to resolve with the given path.
+        def possible_files(context, path, base_path)
+          paths, root_path = super(context, path, base_path)
+          paths = additional_paths_for_sprockets(context, paths, path, base_path)
+          [paths.compact, root_path]
+        end
+
+        def additional_paths_for_sprockets(context, paths, path, base_path)
+          relatives = paths.map { |path_detected| path_detected.to_s.start_with?('.') ? Pathname.new(path_detected) : Pathname.new(path_detected.to_s.prepend('./')) }
+          context.environment.paths.each do |load_path|
+            relative_path = Pathname.new(base_path).relative_path_from(Pathname.new(load_path)).join(path)
+            paths.unshift(relative_path, partialize_path(relative_path))
+          end
+          paths = paths.unshift(relatives)
+          paths
+        end
+
+        def filtered_processor_classes
+          classes = super
+          classes << Sprockets::Preprocessors::DefaultSourceMap if defined?(Sprockets::Preprocessors::DefaultSourceMap)
+          classes << Sprockets::SourceMapProcessor if defined?(Sprockets::SourceMapProcessor)
+          classes
+        end
+
+        def content_type_of_path(context, path)
+          attributes = context.environment.send(:parse_path_extnames, path.to_s)
+          content_type = attributes[1]
+          [content_type, attributes]
+        end
+
+        def build_input_for_process(context, path, data)
+          content_type, _attributes = content_type_of_path(context, path)
+          {
+            environment: context.environment,
+            cache: context.environment.cache,
+            uri: path.to_s,
+            filename: path.to_s,
+            load_path: context.environment.paths,
+            name: File.basename(path),
+            content_type: content_type,
+            data: data,
+            metadata: context.metadata
+          }
+        end
+
+        def check_path_before_process(_context, path)
+          path
+        end
+
+        def call_processor_input(processor, context, input, processors)
+          metadata = (input[:metadata] || {}).dup
+          metadata[:data] = input[:data]
+          result = processor.call(input)
+          processors.delete(processor)
+          handle_process_result(context, result, processors, metadata)
+        end
+
+        def handle_complex_process_result(context, result, processors)
+          data = result[:data] if result.key?(:data)
```

Comment:
Should this be `result.key?(:required)`? Maybe this can be factored into the next if.